Library/ViewModels/PledgePaymentMethodsViewModel.swift

```diff
@@ -227,16 +224,14 @@ public final class PledgePaymentMethodsViewModel: PledgePaymentMethodsViewModelT
             )
           }
 
-          let usePaymentIntent = paymentSheetType == .paymentIntent
+          let usePaymentIntent: Bool = paymentSheetType == .paymentIntent
+          let selectedPaymentMethod: PaymentSourceSelected = usePaymentIntent ?
+            .paymentIntentClientSecret(clientSecret) : .setupIntentClientSecret(clientSecret)
 
-          let selectionUpdatedData = updatedData
-            |> \.paymentMethodsCellData .~ cellData(data.paymentMethodsCellData, selecting: nil)
-            |> \.paymentSheetPaymentMethodsCellData .~ selectedAllSheetPaymentMethods
-            |> \.selectedPaymentMethod .~
-            (
-              usePaymentIntent ? .paymentIntentClientSecret(clientSecret) :
-                .setupIntentClientSecret(clientSecret)
-            )
+          var selectionUpdatedData = updatedData
```

Comment:
One thing I want to poke at a bit while we're talking about deprecating Prelude - this pattern does lose us the ability to use `let` here (like `let selectionUpdatedData = updatedData...`), but because `selectionUpdatedData` is a struct, it effectively does the same thing. We're still creating a new copy of `updatedData` with a couple mutated properties, and we're doing it in substantially less code.

Thoughts?

Reply:
cc @ifosli since you're not on the review